packages/doenetml-worker-rust/lib-doenetml-core/src/dep_graph/graph.rs

```diff
@@ -0,0 +1,307 @@
+use std::{collections::HashMap, fmt::Debug, hash::Hash};
+
+/// An abstract graph. The speed of lookups is determined by the `Taggable` type,
```

Comment:
Somewhere in the comments or even the name of the data structure, it should make it clear we are taking about a directed graph.

Reply:
Maybe called it a "directed acyclic graph" or "DAG"

Reply:
Could be cyclic, so I went with `DirectedGraph`

Reply:
Well, it can't be cyclic, or your walk functions panic!

Reply:
But, on purpose, no checks are done when edges are inserted, so you could make a cycle, just not query a lot from it.

Reply:
Fair enough!